src/gui/dialogs/language_selection.cpp

```diff
@@ -62,9 +126,15 @@ void language_selection::post_show(window& window)
 
 		assert(res != -1);
 
-		const std::vector<language_def>& languages = get_languages();
-		::set_language(languages[res]);
-		preferences::set_language(languages[res].localename);
+		if(langs_[res] == get_language()) {
```

Comment:
I'd invert this condition and put the actions in the block.

Reply:
Oops. This wasn't even meant to be in the PR yet. Noted though.